Application/components/FriendsColumn/AddFriendModal.tsx

```diff
@@ -4,24 +4,41 @@ import { Modal, Tooltip, ActionIcon, Text, Stack, Button, TextInput, useMantineT
 import { IconPlus } from '@tabler/icons-react';
 import { NewFriendModalProps } from '@/accordTypes';
 import { notifications, showNotification } from '@mantine/notifications';
+import { useUser } from '@clerk/nextjs';
 
 export function AddFriendModal({ senderID, setLastFetched }: NewFriendModalProps) {
+  const { user } = useUser();
+
   const [opened, { open, close }] = useDisclosure(false);
+  const [myUsername, setMyUsername] = useState<string>('');
   const [friendUsername, setFriendUsername] = useState('');
   const [searchResult, setSearchResult] = useState<number | null>(null);
   const [errorMessage, setErrorMessage] = useState('');
 
   const theme = useMantineTheme();
 
+  useEffect(() => {
+    if (user && user.username) {
+      setMyUsername(user.username);
+    }
+  }, [user]);
+
     /**
    * Handles the logic for sending a friend request based on the entered username. It performs a POST
    * request to the server with the sender's ID and the friend's username. Upon a successful request,
    * it resets the input field for further requests and triggers a re-fetch of the friend list by updating
    * the `lastFetched` state. If the request fails or if no username is entered, it sets an appropriate
    * error message.
    */
-  const handleAddFriendClick = async () => {
-    if (friendUsername.trim()) {
+    const handleAddFriendClick = async () => {
```

Comment:
- The validation logic in handleAddFriendClick can be improved. Instead of relying on a single errorMessage state, you could define specific error states for different types of errors (e.g., empty username, self-addition). This allows for more precise error messaging and user feedback.
Consider providing visual feedback to users when an error occurs, such as displaying the error message next to the input field or using a toast/notification component.